### PR TITLE
The 46 47 49

### PR DESCRIPTION
**Pull request by MaksymilianK**

### CODE REVIEW

**Comment by elacymerys** [comment]
Chosen categories are not removed from localStorage after signing out, so when you log in again the previously chosen ones are pre-chosen (even if you then log in as different user)

**Comment by elacymerys** [comment]
Is there a reason to have .ts files instead of .tsx ones?

File: thesis-gui/src/services/http-service.tsx [hunk 1]
```diff
@@ -1,20 +1,37 @@
-import axios, { AxiosInstance } from "axios";
+import {HttpStatusCode} from "../utils/http-status-code";
 
-class HttpService {
-    private static axiosInstance: AxiosInstance = axios.create({
-        baseURL: 'http://127.0.0.1:8080/api'
-    });
+export const httpService = {
+    request(path: string, method: string, body?: any) {
+        return fetch(
```
Comment (elacymerys) [inline thesis-gui/src/services/http-service.tsx:5]: Where does this change from axios to fetch come from?

Comment (MaksymilianK) [reply thesis-gui/src/services/http-service.tsx:5]: I couldn't make axios work with credentials in cookies. I don't know why. What's more, fetch is a modern feature and is natively supported by every browser engine.

**Review by Lozoko — COMMENTED** [review]
Backend side everything looks fine but I spotted one thing. Term Service **depends on** Setup Service and should be initialized after Setup Service finished initialization. In other words, we need to wait for categories in database before we try to update terms. This is most likely the reason why Szymon noticed that we need two boot runs to run the application in case we cleared database.

File: thesis-gui/src/services/http-service.tsx (hunk 1 above)
Comment (elacymerys) [reply thesis-gui/src/services/http-service.tsx:5]: Ok

File: thesis-gui/src/components/question/DefinitionQuestion.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    useEffect(() => {
+        if (chosenCategories.length === 0) {
+            history.push('/categories');
+            return;
+        }
+        getNewQuestion(2);
+    }, [chosenCategories, history]);
+
+    useEffect(() => console.log(3), []);
```
Comment (MaksymilianK) [inline thesis-gui/src/components/question/DefinitionQuestion.tsx]: I forgot to remove these console.log. I'll remove them before merging.

**Review by elacymerys — APPROVED** [review]

**Pull request merged**